db/engine.py

```diff
@@ -28,5 +31,33 @@ def create_future_engine(
     return create_engine(conn_str, *args, **kwargs)
 
 
-def _add_custom_types_to_engine(engine):
-    engine.dialect.ischema_names.update(types.CUSTOM_TYPE_DICT)
+def create_engine(conn_str, *args, **kwargs):
+    """
+    Wrapper over sqlalchemy.create_engine that stops SA from propagating changes to ischema_names
+    across all engines. This is important for testing: without this intervention, fixtures become
+    randomly corrupted.
+    """
+    engine = sa_create_engine(conn_str, *args, **kwargs)
+    _make_ischema_names_unique(engine)
+    return engine
+
+
+# TODO should refactor for this to be private
```

Comment:
I try to keep things used in testing public. My intention here is to refactor it so that we never call it directly. Instead I'd like to be called indirectly when creating an engine.